pbr/config/scene_config.py

```diff
@@ -191,24 +192,47 @@ def configure_scene():
             ]
         }
     )
+
     # Add robot information
+    xbound = (2 * cfg["field"]["border_width"] + cfg["field"]["length"]) / 2
+    ybound = (2 * cfg["field"]["border_width"] + cfg["field"]["width"]) / 2
+    # Let the robot's personal space be a tenth of the width. (Should be added to config somewhere)
+    r = cfg["field"]["width"] / 5
+
+    # This recursive function makes sure that the robots will not overlap
+    def _generate_positions(
+        num_robots,
+        radius,
+        bounds=[(-xbound, xbound), (-ybound, ybound)],
+        positions=[],
+    ):
+        # Unpack bounds and assign names for clarity
+        xbounds, ybounds = bounds
+        xmin, xmax = xbounds
+        ymin, ymax = ybounds
+
+        if len(positions) == num_robots:
+            return positions
+
+        else:
+            xnew, ynew = np.random.uniform(xmin, xmax), np.random.uniform(ymin, ymax)
+            if any(
+                (xnew - p[0]) ** 2 + (ynew - p[1]) ** 2 < radius**2 for p in positions
+            ):
+                return _generate_positions(num_robots, radius, bounds, positions)
+            else:
+                positions.append((xnew, ynew))
+                return _generate_positions(num_robots, radius, bounds, positions)
```

Comment:
This could do with more commenting. Using recursion in this case makes for unnecessarily complex code, I would probably just use loops.

Reply:
Soz, wanted to avoid the nasty edge cases and was very lazy so I just did this. Will make it cleaner